core/new-gui/src/app/common/service/user/user.service.ts

```diff
@@ -46,7 +47,10 @@ export class UserService {
   }
 
   public logout(): void {
-    this.authService.logout().subscribe(_ => this.changeUser(undefined));
+    this.authService.logout().subscribe(_ => {
+      this.changeUser(undefined);
+      this.router.navigate(["login"]);
```

Comment:
we should not do re-routing inside `user-serivce`. The scope of user service is clear: it manages the current user information of the system, and navigation is NOT its task.

There should be another service (or component, if applicable) to listen to and handle `userChanged` event, so that it can navigate accordingly.